sentry/src/main/java/io/sentry/SentryIntegrationPackageStorage.java

```diff
@@ -64,12 +67,48 @@ public void addPackage(final @NotNull String name, final @NotNull String version
 
     SentryPackage newPackage = new SentryPackage(name, version);
     packages.add(newPackage);
+    try (final @NotNull ISentryLifecycleToken ignored = mixedVersionsLock.acquire()) {
+      mixedVersionsDetected = null;
+    }
   }
 
   public @NotNull Set<SentryPackage> getPackages() {
     return packages;
   }
 
+  public boolean checkForMixedVersions(final @NotNull ILogger logger) {
+    final @Nullable Boolean mixedVersionsDetectedBefore = mixedVersionsDetected;
+    if (mixedVersionsDetectedBefore != null) {
+      return mixedVersionsDetectedBefore;
+    }
+    try (final @NotNull ISentryLifecycleToken ignored = mixedVersionsLock.acquire()) {
+      final @NotNull String sdkVersion = BuildConfig.VERSION_NAME;
+      boolean mixedVersionsDetectedThisCheck = false;
+
+      for (SentryPackage pkg : packages) {
+        if (pkg.getName().startsWith("maven:io.sentry:")
+            && !sdkVersion.equalsIgnoreCase(pkg.getVersion())) {
+          logger.log(
+              SentryLevel.ERROR,
+              "The Sentry SDK has been configured with mixed versions. Expected %s to match core SDK version %s but was %s",
+              pkg.getName(),
+              sdkVersion,
+              pkg.getVersion());
+          mixedVersionsDetectedThisCheck = true;
+        }
+      }
+
+      if (mixedVersionsDetectedThisCheck) {
+        logger.log(SentryLevel.ERROR, "^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+        logger.log(SentryLevel.ERROR, "^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+        logger.log(SentryLevel.ERROR, "^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+        logger.log(SentryLevel.ERROR, "^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+      }
+      mixedVersionsDetected = mixedVersionsDetectedThisCheck;
```

Comment:
shall this not be inside the locking block? If someone adds a new package in-between line 97 and 107 are executed, this will hold a wrong value right?

Reply:
Hmm I don't understand, the `try` block ends on line 109

Reply:
oh, nvm, I did not see that 🙈